utils/cloudinfo/gce.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudinfo
+
+import (
+	"io"
+	"net/http"
+	"strings"
+
+	info "github.com/google/cadvisor/info/v1"
+)
+
+func inGCE() bool {
+	_, err := http.Get("http://metadata.google.internal/computeMetadata/v1/instance/machine-type")
```

Comment:
I recommend using: https://godoc.org/google.golang.org/cloud/compute/metadata

It's pretty simple code, but useful to use a library by the authors/maintainers :)